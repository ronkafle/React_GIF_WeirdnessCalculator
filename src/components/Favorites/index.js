import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory } from "react-router-dom";
import Thumbnail from '../Thumbnail';
import { Col, Row, Container, Button } from 'react-bootstrap';
import { MIN_LIKED_GIFS } from '../../configs/constants';
import { FAVORITES } from '../../actions/favorites';
import { SEARCH } from '../../actions/search';

const Favorites = () => {

  const favorites = useSelector(state => state.favorites);
  const query = useSelector(state => state.search.query);  
  const gifsNeeded = MIN_LIKED_GIFS - Object.keys(favorites).length;
  const dispatch = useDispatch();
  const history = useHistory();

  const onRemove = (term) => {
    dispatch({
      type: FAVORITES.REMOVE,
      payload: {
        term,
      }
    });
    if (term === query) {
      dispatch({
        type: SEARCH.LIKED,
        payload: false
      });
    }
  };

  const calculateScore = (e) => {
    history.push('/result');
  };

  return (
    <Container className="favorites">
      <h2>YOUR LIKED GIFS</h2>
      <Row>
        {Object.keys(favorites).map((f, i) => (
          <Col key={i}>
            <Thumbnail
              title={favorites[f].result.title}
              src={favorites[f].result.images.original.url}
              onCancel={() => onRemove(f)} />
          </Col>
        ))}
      </Row>
      <Row>
        <Col className="favorites-footer">
          <Button onClick={calculateScore} disabled={gifsNeeded} className="btn btn-primary">
            Calculate My Weirdness Score
        </Button>
          {
            gifsNeeded > 1
              ? <p>You must <i>Like</i> {`${gifsNeeded} more GIF${gifsNeeded > 1 ? 's' : ''}`} to calculate your score</p>
              : null
          }
        </Col>
      </Row>
    </Container>
  );
};

export default Favorites;
