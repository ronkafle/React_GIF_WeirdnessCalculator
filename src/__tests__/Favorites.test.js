import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Favorites from '../components/Favorites';
import configureStore from "../store";

const store = configureStore();

describe('Component: Favorites', () => {
  let component = null;
  
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <Favorites />
      </Provider>
    );
  });

  it('renders without crashing', () => {
    expect(component).toMatchSnapshot();
  });
});



