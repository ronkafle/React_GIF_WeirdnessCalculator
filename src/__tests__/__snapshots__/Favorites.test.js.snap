// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Favorites renders without crashing 1`] = `ShallowWrapper {}`;
